ecstats.py

```diff
@@ -153,18 +153,29 @@ def create_workbook(outDir, section, region_name):
     ws = wb.active
     ws.title = RUNNING_INSTANCES_WORKSHEET_NAME
 
-    df_columns = ["ClusterId", "NodeId", "NodeType", "Region"]
+    df_columns = ["Source", "ClusterId", "NodeId", "NodeRole", "NodeType", "Region"]
     for metric, _, _ in get_max_metrics_weekly():
-        df_columns.append(('%s (max over last week)' % metric))
+        df_columns.append(metric)
     for metric, _, _ in get_max_metrics_hourly():
-        df_columns.append(('%s (peak last week / hour)' % metric))
+        df_columns.append(metric)
     ws.append(df_columns)
 
     ws = wb.create_sheet(RESERVED_INSTANCES_WORKSHEET_NAME)    
     df_columns = ["Instance Type", "Count", "Remaining Time (days)"]
     ws.append(df_columns)    
     return wb
 
+def get_instance_role(clusterId, nodeId):
```

Comment:
def get_instance_role(clusterId, nodeId):
    parts = nodeId.replace(clusterId, "").strip("-").split("-")
    if len(parts) == 1 or  clusterId == nodeId or int(parts[1]) == 1 :
        return "Master":
    return "Replica"

Reply:
so you prefer less IFs and more complex logic? I always try to make the code more easy to read, but this is my style :)